features/steps/cli_steps.py

```diff
@@ -242,6 +242,25 @@ def create_project_with_starter(context):
     assert res.returncode == OK_EXIT_CODE, res
 
 
+@given("I have run a non-interactive kedro new with custom plugin starter")
+@when("I run a non-interactive kedro new with custom plugin starter")
+def create_project_with_plugin_starter(context):
+    """Behave step to run kedro new given the config I previously created."""
```

Comment:
Maybe we can merge this with `create_project_with_starter` above and pass an optional argument with a starter name. If it's not provided then it would just use `starter_dir = Path(__file__).parent / "test_starter"`